### DISCUSSION TITLE
Copy effects to multiple targets

### DISCUSSION THREAD

**Discussion by RafaelLinux** [olive-editor/olive#2269] [Q&A]
I know is possible to copy effect (one by one) to another clip using nodes. But I have two questions:

1. Is possible to copy more than one effect at the same time to another clip?
2. Is possible to copy at least one effect to multiple clips?

Thank you

**Comment by AngeloFrancabandiera** [comment]
Not sure I understand your question, but you can copy a set of nodes from one clip to another. Just select the block of nodes by click & drag and paste it to the target node. You can paste the block in all the clips that you want, but not in a single paste operation (AFAIK).

**Comment by RafaelLinux** [comment]
Well, yes, I was asking about to paste to selected clips in a single operation.
So it's not possible, thanks. 
I think that feature is "a must". Suppose you have about 30 clips, and need to apply them all some efffects ... that's scenario.

**Comment by AngeloFrancabandiera** [comment]
A possible workaround (with some limitation) is sequence nesting. Put all the clips that you want to apply the effect to in a sequence. Then create a new sequence and put the first sequence in the timeline of the second as with any other footage. In the external sequence you see the internal one as a clip and you can apply an effect that will be applied to all clips of the internal sequence.
You can still trim the clips inside the internal sequence later and this will reflect in the external sequence.

**Comment by RafaelLinux** [comment]
I tried something similar, but didn't work as expected. Let me explain. I have a series of pictures (N pictures) and want to make a slide show, so each one will have its own zoom, rotation and movement animations. However, they all will have same white frame around and a shadow. So these effect are common to all them. In my experiment, Olive finally considered all photos (when overlap one over others) same object, so frame and shadow was not for each picture but for all as a unique object.

However, if I make nodes like in attached screenshot, it works fine, but this method forces me to repeat N times the shadow, outline effects and to have to insert N-1 "join" nodes. And all this job "manually" having to link at least 2 * N times nodes.
![imagen](https://github.com/olive-editor/olive/assets/13048959/c2840dbf-3c9b-4154-9ac4-2fed1dc52751)

**Comment by AngeloFrancabandiera** [comment]
I've been thinking a while; I don't think there's an easy solution. Even if there was a way (and there is not) to paste the "border+drop_sahdow" block in each clip, this block would not be automatically connected just before the output node. The effort to go into each clip and connected the pasted nodes to the output is not quite different to also paste the block that you keep on the clipboard.

**Reply by RafaelLinux** [reply]
If exist an easy way to change picture of a clip yet in timeline for a picture available in media library, user could duplicate the clip with all effects and simply select a duplicated clip, select a picture from media, and select "change content" or "change input source by selected media" or something similar.
